ethtx/providers/semantic_providers/database.py

```diff
@@ -121,3 +98,18 @@ def insert_address(self, address, update_if_exist=False) -> Optional[bson.Object
 
         inserted_address = self._addresses.insert_one(address_with_id)
         return inserted_address.inserted_id
+
+    def _init_collections(self) -> None:
+        if MongoCollections.ADDRESSES not in self._db.list_collection_names():
+            self._addresses = self._db[MongoCollections.ADDRESSES]
+
+        if MongoCollections.CONTRACTS not in self._db.list_collection_names():
+            self._contracts = self._db[MongoCollections.CONTRACTS]
+
+        if MongoCollections.SIGNATURES not in self._db.list_collection_names():
+            self._signatures = self._db[MongoCollections.SIGNATURES]
+            self._signatures.create_index(
+                [("signature_hash", "TEXT"), ("name", "TEXT")],
+                background=True,
+                unique=False,
+            )
```

Comment:
1. Take list_collection_names and put it in variable, instead of calling it 3 times.
2. Why are those variables not initialized when collections are not in db?

Reply:
and unique shouldn't be True?

Reply:
Better, but still if collections are not in db, `self._addresses` `self._contracts` and `self._signatures` will be empty. Normally those collections would be created by mongo itself, here program will just fail if those are not created by user manually.